src/Proxy.hs

```diff
@@ -1698,65 +1756,70 @@ instance FromJSON DropSchedule where
       return Normal
     else fail "Unsupported drop type."
 
--- | Handle a GTU drop request.
---     When 'gtuDropData' is provided: handle drop.
---     Otherwise: return 404.
+-- |Return a handler which GTU drop.
+-- Attempt to make a GTU drop to the account with the specified address if @gtuDropData@ was provided
+-- in the environment settings. Responds with a 404-status code if @gtuDropData@ was absent.
 putGTUDropR :: Text -> Handler TypedContent
 putGTUDropR addrText = do
-    i <- internationalize
     Proxy{..} <- getYesod
     case gtuDropData of
+      -- VH/FIXME: Should this be 404?
       Nothing -> respond404Error $ EMErrorResponse NotFound
       Just gtuDropData' -> do
-        case addressFromText addrText of
-          Left _ -> respond400Error EMMalformedAddress RequestInvalid
-          Right addr -> runGRPC (doGetAccInfo addrText) $ \case
-              -- Account is not finalized
-              Nothing -> sendResponseStatus notFound404 $ object
-                          ["errorMessage" .= i18n i EMAccountNotFinal,
-                          "error" .= fromEnum RequestInvalid]
-              -- Account is finalized, so try the drop
-              Just _ -> do
-                connect rawRequestBody (sinkParserEither json') >>= \case
-                  Left _ -> tryDrop addr Normal gtuDropData' -- malformed or empty body, we assume normal drop.
-                  Right v -> case fromJSON v of
-                              AE.Success x -> tryDrop addr x gtuDropData'
-                              AE.Error e -> do
-                                $(logWarn) (Text.pack e)
-                                respond400Error EMConfigurationError RequestInvalid
-
+        addr <- doParseAccountAddress (Just "putGTUDrop") addrText
```

Comment:
Should we check some notion of the account or its transactions being finalized here? It seems something like that was done before, but I am not quite sure of the semantics of the old API, so I can not say for sure. In that case, could this be done by using `getNextSequenceNumber` and asserting `nanAllFinal` and aborting otherwise?

Reply:
@abizjak